pkg/models/terminal/terminal.go

```diff
@@ -140,15 +149,133 @@ func (t TerminalSession) Close(status uint32, reason string) {
 
 type Interface interface {
 	HandleSession(shell, namespace, podName, containerName string, conn *websocket.Conn)
+	HandleShellAccessToNode(nodename string, conn *websocket.Conn)
 }
 
 type terminaler struct {
-	client kubernetes.Interface
-	config *rest.Config
+	client  kubernetes.Interface
+	config  *rest.Config
+	options *Options
+}
+
+type NodeTerminaler struct {
+	Nodename      string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Shell         string
+	Privileged    bool
+	Config        *Options
+	client        kubernetes.Interface
 }
 
-func NewTerminaler(client kubernetes.Interface, config *rest.Config) Interface {
-	return &terminaler{client: client, config: config}
+func NewTerminaler(client kubernetes.Interface, config *rest.Config, options *Options) Interface {
+	return &terminaler{client: client, config: config, options: options}
+}
+
+func NewNodeTerminaler(nodename string, options *Options, client kubernetes.Interface) (*NodeTerminaler, error) {
+
+	n := &NodeTerminaler{
+		Namespace:     "kubesphere-controls-system",
+		ContainerName: "nsenter",
+		Nodename:      nodename,
+		PodName:       nodename + "-shell-access",
+		Shell:         "sh",
+		Privileged:    true,
+		Config:        options,
+		client:        client,
+	}
+
+	node, err := n.client.CoreV1().Nodes().Get(context.Background(), n.Nodename, metav1.GetOptions{})
+
+	if err != nil {
+		return n, fmt.Errorf("node cannot exist. nodename:%s, err: %v", n.Nodename, err)
```

Comment:
`Nodes().Get()` returns err doesn't always means `node not exist`, sometimes it's network issue or internal error, so I suggest change the error message to
```suggestion
		return n, fmt.Errorf("getting node error, nodename:%s, err: %v", n.Nodename, err)
```